src/base.jl

```diff
@@ -751,3 +785,17 @@ function compare_values(x::System, y::System)::Bool
 
     return match
 end
+
+function read_validation_descriptor(filename::AbstractString)
+    if !occursin(r"(\.yaml)|(\.yml)"i, filename)
+        error("Must enter a path to YAML file for configpath kwarg in System struct")
```

Comment:
It would be better to have an error message that applies to this function, not the one calling it, i.e., "filename is not a YAML file."